src/components/SearchForm.tsx

```diff
@@ -21,26 +22,25 @@ export function SearchForm({
 	const [selectedWiki, setSelectedWiki] = useState<WikiSite>(searchState.wiki);
 	const [pageId, setPageId] = useState<number | null>(null);
 	const [pageIdError, setPageIdError] = useState<string | null>(null);
-	const debouncedPageTitle = useDebounce(pageTitle, 300);
+	const debouncedPageTitle = useDebounce(pageTitle.trim(), 300);
 
-	useEffect(() => {
+	const pageIdPromise: Promise<void> = useMemo(async () => {
+		// reset
 		setPageId(null);
 		setPageIdError(null);
-
+		// guard
 		if (!debouncedPageTitle) {
 			return;
 		}
-
-		fetchPageId(selectedWiki.url, debouncedPageTitle)
-			.then((id) => {
-				setPageId(id);
-			})
-			.catch((error) => {
-				console.error('Error fetching page ID:', error);
-				setPageIdError('Error fetching page ID. Please try again.');
-				setPageId(null);
-			});
-	}, [debouncedPageTitle, selectedWiki]);
+		// fetch page ID
+		try {
+			const id = await fetchPageId(selectedWiki.url, debouncedPageTitle);
+			setPageId(id);
+		} catch (error) {
+			console.error('Error fetching page ID:', error);
+			setPageIdError('Error fetching page ID. Please try again.');
+		}
+	}, [selectedWiki, debouncedPageTitle]);
```

Comment:
It won't work if we bring the promise inside the child component to be suspended somehow.

Reply:
Such a promise gets discarded on every suspend.